Donation-platform/src/main/java/com/google/servlets/PersonalizationServlet.java

```diff
@@ -38,30 +39,25 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Parse tags JSON from request into a list of tags
     List<String> tags = new ArrayList<>();
-
-    // Get the user's tag input from the checkbox form
-    String tag1 = request.getParameter("tag-1");
-    String tag2 = request.getParameter("tag-2");
-    String tag3 = request.getParameter("tag-3");
-    tags.add(tag1);
-    tags.add(tag2);
-    tags.add(tag3);
+    // requestData example: {"tag1":"hunger","tag2":"education","tag3":"children"}
+    String requestData = request.getReader().lines().collect(Collectors.joining());
```

Comment:
IIUC, we are fetching tags value from this json string. I think we should convert this json string into key value pair in Java and then get all values instead of parsing the string this way. Check https://www.javainterviewpoint.com/convert-json-java-map-jackson/

Reply:
Done.